module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,70 @@
+#include "compression.hpp"
+
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+
+CompressedPGMBitMap compressGrayscale(const PGMBitMap& img) {
+    CompressedPGMBitMap bitMap;
+    bitMap.reserve(height);
```

Comment:
You reserve not enough memory. In worst case you will need height * width size. Also use `shirnk_to_fit`

Reply:
That was intentional - so i don't reserve too much memory. (and because i don't do that i don't need shrink_to_fit here).